util/users.py

```diff
@@ -55,3 +55,21 @@ async def delete_user(identifier: user_models.UserIdentifier) -> None:
     if response.deleted_count == 0:
         detail = "User not found and could not be deleted"
         raise exceptions.UserNotFoundException(detail=detail)
+
+
+async def validate_user_info(
+        validate: user_models.ValidateUserForm) -> str:
+    """
+    Validates user based off identifier and password
+    #todo: call get_user_info_by_identifier w identifier, validate pass vs user object. return 200
+    """
+    try:
+        user_info = await get_user_info_by_identifier(validate.identifier)
+    except exceptions.UserNotFoundException:
+        return 'not found'
+
+    # if user_info.hashed_password == hash_pass(validate.password):
+    if validate.password == 'password123':
+        return 'token here'
+    else:
+        return 'not found'
```

Comment:
This is obviously placeholder code, but it will not be easy or clean to refactor.

A better approach would be offloading the password comparison to a function so _when_ we need to actually implement it, this code will be the exact same.

e.g.
```python
password_matches = check_user_password_matches(login_form, user_info) # should return bool
if password_matches:
    return get_auth_token_from_user_data(user_data)
else:
    raise exceptions.InvalidPasswordException # this doesn't exist yet but it should throw a 422 or 403
```

This way, when we want to change anything about password comparison we can easily just modify the `check_user_password_matches` function. This is the core of making clean code: how easy it is to refactor for improvement.

Reply:
Isn't it bad practice to return a different response when the password is wrong versus when the user doesn't exist?

Reply:
Agreed, _returning_ something different is bad, but we're actually not returning something different if the password is invalid, we're raising an exception. 

Only the "successful" control flow should return an object, all others should just raise exceptions (they go all the way to the top of the stack to the client).